Add f-empty? and alias.

mwfogleman opened the pull request:
I needed this function and didn't find it, and thought it might be useful to others.

I realize this isn't in the predicates section, but the function it uses - f-size - comes later in the file. Any thoughts about that? Is there a better implementation?

rejeep commented:
Makes sense. Can you please add some docs and specs. Also move it to the predicates section.

mwfogleman commented:
Sure, I'm happy to add some docs. If the function that it calls, f-size, is later in the file, does that matter in ELisp? I know in Clojure you can't call functions that come later in the file.

rejeep commented:
I don't think it's an issue unless it's a macro. Either way, if you write some specs it should catch that if that's the case.

mwfogleman commented:
OK - how do I write specs? :)

rejeep commented:
There are tons of tests already so you should be able to figure it out by looking at them. Take a look at https://github.com/rejeep/f.el/blob/master/test/f-predicates-test.el for example. Also read this section: https://github.com/rejeep/f.el#contribution

mwfogleman commented:
I moved the functions, added a docstring, and wrote some tests, but I don't think they're working. I'll try to chew on this but assists are welcome :)

rejeep commented on test/f-predicates-test.el:
```diff
@@ -416,6 +416,19 @@
+(ert-deftest f-empty?-test/file-containing-text-is--not-empty ()
+  (with-playground
+   (f-touch "derp.txt")
+   (f-append-text "hello"' utf-8 "derp.txt")
```
You can use `f-write-text` instead of `f-touch` and `f-append-text`

rejeep commented on test/f-predicates-test.el:
```diff
@@ -215,27 +215,27 @@
    (f-mkdir (f-expand "bar" f-test/playground-path))
    (should-not (f-same? (f-expand "foo" f-test/playground-path)
-                        (f-expand "bar" f-test/playground-path)))))
+			(f-expand "bar" f-test/playground-path)))))
```
Opps, looks like you changed some whitespace here and above

mwfogleman commented:
Ah, great suggestions! I didn't think about directories because I just needed files for my program :)

I've implemented and tested that functionality - let me know if you think there's a better way of checking if the directory is empty or not. :)

rejeep commented on test/f-predicates-test.el:
```diff
@@ -416,6 +416,29 @@
+  (with-playground
+   (f-mkdir "bar")
+   (f-write-text "hello" 'utf-8 "bar/derp.txt")
+   (should-not (f-empty? "bar/derp.txt"))))
```
I guess you want to do `(should-not (f-empty? "bar"))` here?

mwfogleman commented:
Do you want me to squash these commits?

rejeep commented:
Sure go ahead and squash and then I will merge

mwfogleman commented:
Squashed! Thanks for your help - I learned a little bit about how to write tests in Emacs :) Cheers!

rejeep commented:
One last thing, can you also update the readme template (https://github.com/rejeep/f.el/blob/master/README.md.tpl) and then run `make docs`.

mwfogleman commented:
Sure, so, I added something based on what I saw in that template. However when I run make docs, I get this error:
cask exec emacs -Q --script bin/docs.el
Cannot open load file: No such file or directory, s
make: *** [Makefile:16: docs] Error 255

rejeep commented:
You have to install Cask and run `cask install` first.

rejeep commented:
Also, the docs should be in two places. Look at the other examples in the file.

mwfogleman commented:
Cool, thanks! Happy to squash this if it looks good to you.

rejeep commented:
Lovely and I'll merge!

mwfogleman commented:
Done! :)

rejeep commented:
Great stuff, thanks for contributing! 👍
